spec/article_json/export/amp/exporter_spec.rb

```diff
@@ -9,4 +9,29 @@
     before { stub_oembed_requests }
     it { should eq html.strip }
   end
+
+  describe 'amp_libraries' do
+    subject { exporter.amp_libraries }
+
+    let(:json) { File.read('spec/fixtures/reference_document_parsed.json') }
+    let(:elements) { ArticleJSON::Article.from_json(json).elements }
+    let(:expected_result) do
+      [
+        '<script async custom-element="amp-vimeo"' \
+        'src="https://cdn.ampproject.org/v0/amp-vimeo-0.1.js"></script>',
+        '<script async custom-element="amp-youtube" ' \
+        'src="https://cdn.ampproject.org/v0/amp-youtube-0.1.js"></script>',
+        '<script async custom-element="amp-facebook" ' \
+        'src="https://cdn.ampproject.org/v0/amp-facebook-0.1.js"></script>',
+        '<script async custom-element="amp-iframe"' \
+        'src="https://cdn.ampproject.org/v0/amp-iframe-0.1.js"></script>',
+        '<script async custom-element="amp-twitter" ' \
+        'src="https://cdn.ampproject.org/v0/amp-twitter-0.1.js"></script>'
+      ]
```

Comment:
is the order in a Set defined?

Reply:
Nope:
> Set implements a collection of unordered values with no duplicates.
https://ruby-doc.org/stdlib-2.4.2/libdoc/set/rdoc/Set.html (very first line :wink: )

Reply:
then one should probably not compare these two arrays with ```eq``` below.

Reply:
right, `match_array` is what we should use.

Reply:
changed, thanks!